tcl/handlers.tcl

```diff
@@ -55,29 +55,38 @@ namespace eval qc::handlers {
                 set form_dict [dict merge $form_dict $args]
                 set args [args_from_dict $form_dict $pattern $method]
                 set command [list $pattern {*}$args]
-                if {$colon_vars} {
+                if {$colon_variable_count > 0} {
                     # The proc contains colon variables
-                    dict set pattern_dict variable $command
+                    if { [info exists command_dict] && [dict exists $command_dict variable]} {
+                        # There is currently a colon variable command in the dictionary...
+                        if { $colon_variable_count < [dict get $command_dict variable colon_variable_count] } {
+                            # Prefer the command with the lower number of colon variables
+                            dict set command_dict variable [list command $command colon_variable_count $colon_variable_count]
+                        }
+                    } else {
+                        # No colon variable command currently stored.
+                        dict set command_dict variable [list command $command colon_variable_count $colon_variable_count]
+                    }                    
                 } else {
                     # The pattern was an exact match
-                    dict set pattern_dict exact $command
+                    dict set command_dict exact [list command $command]
                     break
```

Comment:
I can see that this was of setting the dict is consistent with earlier code when you were setting a dictionary value to be another dict.

Just an FYI that you can do this with just a dict set:

```
dict set command_dict exact command $command
```